### PR TITLE
conform: init at 0.1.0-alpha.25

### PR DESCRIPTION
**Pull request by jmgilman**
###### Description of changes

Adds a package for building [conform](https://github.com/siderolabs/conform), a tool for enforcing policies in git commits.  

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/conform/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    repo = "conform";
+    sha256 = "sha256-WgWgigpqPoEBY4tLjbzK02WFwrCWPGQWJ5eakLv5IWw=";
+  };
+
+  vendorSha256 = "sha256-Oigt7tAK4jhBQtfG1wdLHqi11NWu6uJn5fmuqTmR76E=";
+
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Policy enforcement for your pipelines.";
```
Comment (06kellyjac) [inline pkgs/applications/version-management/git-and-tools/conform/default.nix:19]:
```suggestion
    description = "Policy enforcement for your pipelines";
```

contributing mandates no period at the end of description

File: pkgs/applications/version-management/git-and-tools/conform/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "siderolabs";
+    repo = "conform";
+    sha256 = "sha256-WgWgigpqPoEBY4tLjbzK02WFwrCWPGQWJ5eakLv5IWw=";
+  };
+
+  vendorSha256 = "sha256-Oigt7tAK4jhBQtfG1wdLHqi11NWu6uJn5fmuqTmR76E=";
+
+  doCheck = false;
```
Comment (06kellyjac) [inline pkgs/applications/version-management/git-and-tools/conform/default.nix:16]: why?

when I packaged this for my nur this was enough:

`checkInputs = [ git ];`

File: pkgs/applications/version-management/git-and-tools/conform/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    description = "Policy enforcement for your pipelines.";
+    homepage = "https://github.com/siderolabs/conform";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ jmgilman ];
```
Comment (06kellyjac) [inline pkgs/applications/version-management/git-and-tools/conform/default.nix:22]: I'm happy to maintain this with you if you'd like (just add `jk` to the list, or dont)

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
Some small changes, also consider adding the following for a smaller binary and so `conform version` works

```
  ldflags = [
    "-s"
    "-w"
    "-X github.com/talos-systems/conform/internal/version.Tag=v${version}"
  ];
```

**NOTE:** this must stay as `talos-systems` not `siderolabs` until they change this in the go.mod https://github.com/siderolabs/conform/blob/master/go.mod#L1

It's a shame conform's development has stalled

File: pkgs/applications/version-management/git-and-tools/conform/default.nix (hunk 2 above)
Comment (jmgilman) [reply pkgs/applications/version-management/git-and-tools/conform/default.nix]: This was my fault. When I first looked I didn't see any tests.

**Comment by jmgilman** [comment]
> consider adding the following for a smaller binary and so `conform` version work

Would you mind clarifying what's being done here?

**Comment by jmgilman** [comment]
> It's a shame conform's development has stalled

I reached out to the team maintaining it on their Slack. They indicated they use it in all of their projects and they haven't felt the need to add anything additional to it. I'm trying to convince them to remove the `alpha` status to indicate that it's essentially in a working state.

**Comment by 06kellyjac** [comment]
> > consider adding the following for a smaller binary and so `conform` version work
> 
> Would you mind clarifying what's being done here?

Sure. ldflags are flags that are passed to the linker. `-s` & `-w` will strip info only useful for development. `-s` removes the symbol table like `strip -s` does and `-w` disables DWARF debugging info

```
$ go tool link
  ...
  -s    disable symbol table
  -w    disable DWARF generation
```

`-X some/path/to.Value=something` is a way to override `vars` and is commonly used to set values used for version info

https://github.com/siderolabs/conform/blob/master/internal/version/version.go#L17-L22

With `buildGoModule` you can set `ldflags` to be an array of values like listed above

File: pkgs/applications/version-management/git-and-tools/conform/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{ pkgs, lib, buildGoModule, fetchFromGitHub }:
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/conform/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, git }:
```
we don't need to use that here.

File: pkgs/applications/version-management/git-and-tools/conform/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "siderolabs";
+    repo = "conform";
+    sha256 = "sha256-WgWgigpqPoEBY4tLjbzK02WFwrCWPGQWJ5eakLv5IWw=";
+  };
+
+  vendorSha256 = "sha256-Oigt7tAK4jhBQtfG1wdLHqi11NWu6uJn5fmuqTmR76E=";
+
+  checkInputs = [ pkgs.git ];
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/conform/default.nix:16]:
```suggestion
  checkInputs = [ git ];
```

**Review by 06kellyjac — APPROVED** [review]

**Pull request merged**